core/tests/protractor_utils/editor.js

```diff
@@ -1318,6 +1318,43 @@ var revertToVersion = function(version) {
   });
 };
 
+// Wrapper for functions involving the feedback tab
+var _runFromFeedbackTab = function(callbackFunction) {
+  element(by.css('.protractor-test-feedback-tab')).click();
+  var result = callbackFunction();
+  general.waitForSystem();
+  element(by.css('.protractor-test-main-tab')).click();
+  return result;
+};
+
+var readFeedbackMessages = function() {
+  return _runFromFeedbackTab(function() {
+    var feedbackRowClassName = '.protractor-test-oppia-feedback-tab-row';
+    var messages = [];
+    return element.all(by.css(feedbackRowClassName)).then(function(rows) {
+      rows.forEach(function(row) {
+        row.click();
+        element(by.css('.protractor-test-exploration-feedback'))
+          .getText().then(function(message) {
+            messages.push(message);
+          });
+        element(by.css('.protractor-test-oppia-feedback-back-button')).click();
+      });
+      return messages;
+    });
+  });
+};
+
+var sendResponseToLatestFeedback = function(feedbackResponse) {
+  element(by.css('.protractor-test-feedback-tab')).click();
+  element.all(by.css('.protractor-test-oppia-feedback-tab-row')).last().click();
```

Comment:
Just to check -- does the most recent feedback appear on the bottom or the top? I thought it was the top...